firstvoices/backend/views/games_views.py

```diff
@@ -30,6 +33,67 @@ def get_wordsy_solution_seed(num_words):
     return index % num_words
 
 
+def get_wordsy_possible_solutions(site) -> list:
+    """Get friendly words with custom length matching the number of game letters."""
+    words_qs = (
+        # filter words based on custom length; this also filters out unknown chars
+        DictionaryEntry.objects.annotate(
+            custom_char_length=RawSQL("ARRAY_LENGTH(split_chars_base, 1)", ())
+        )
+        # visible, kid-friendly, and games-friendly words only
+        .filter(
+            site=site,
+            visibility=site.visibility,
+            type=TypeOfDictionaryEntry.WORD,
+            exclude_from_games=False,
+            exclude_from_kids=False,
+            custom_char_length__exact=SOLUTION_LENGTH,
+        )
+        # words cannot contain spaces
+        .exclude(title__contains=" ")
+        .exclude(custom_order__contains=UNKNOWN_FLAG)
+        .order_by("custom_order")
+        .distinct("custom_order")
+        .values_list("split_chars_base", flat=True)
+    )
+    solutions = []
+    for split_chars in words_qs.iterator():
+        title_as_base_chars = "".join(split_chars)
+        solutions.append(title_as_base_chars)
+    return solutions
+
+
+def get_wordsy_possible_guesses(site) -> list:
```

Comment:
This is the added section that searches for a different collection of possible guessable words, using different possible splitting methods. It is time consuming (~5sec for ~13000 entries) but the result is cached the first time the endpoint is hit.

If this is too slow, I could swap it out to use the original splitting method which would be much faster. And we could save this option (or a better one) for a future improvement in the async direction, maybe relying on a scheduled job instead.

Reply:
Since the pool of valid guesses only changes if a word is added or edited, I wonder about keeping it up-to-date the same way we deal with the character splitting. If we stored something like is_valid_wordsy_solution and is_valid_wordsy_guess when we saved a word, we'd have to run recalculate dictionary once but then we could retrieve the valid words with a very fast lookup.

Reply:
I didn't think of this, but interesting idea. A possible drawback is that it wouldn't adjust well if we want the number of letters in the Wordsy game to be flexible based on difficulty or language (we would probably want to stick with 5 letters, period).